lib/ts/controllers/s-navigation-tablist.ts

```diff
@@ -0,0 +1,121 @@
+            this.selectedTab?.focus();
+        }
+
+        /**
+         * Handles left and right arrow keydown events on individual tabs,
+         * selecting the adjacent tab corresponding to the event.
+         */
+        handleKeydown(event: KeyboardEvent) {
+            let tabElement = <HTMLElement>event.currentTarget;
+
+            var tabs = this.tabTargets;
+            var tabIndex = tabs.indexOf(tabElement);
+    
+            if (event.key === "ArrowRight") {
+                tabIndex++;
+            } else if (event.key === "ArrowLeft") {
+                tabIndex--;
+            } else {
+                return;
+            }
+    
+            if (tabIndex < 0) { tabIndex = tabs.length - 1; }
+            if (tabIndex >= tabs.length) { tabIndex = 0; }
+
+            tabElement = <HTMLElement>tabs[tabIndex];
+            this.switchToTab(tabElement);
+            this.selectedTab?.focus();
+        }
+
+        /**
+         * Attempts to switch to a new tab, doing nothing if the tab is already selected or
+         * the s-navigation-tablist:select event is prevented.
+         */
+        private switchToTab(newTab: HTMLElement) {
+
+            var oldTab = this.selectedTab;
+            if (oldTab === newTab) { return; }
+
+            if (this.triggerEvent("select", { oldTab, newTab }).defaultPrevented) { return; }
+
+            this.selectedTab = newTab;
+            this.triggerEvent("selected", { oldTab, newTab });
+        }
+        
+        /**
+         * Returns the currently selected tab or null if no tabs are selected.
+         */
+        public get selectedTab() : HTMLElement | null {
+            return this.tabTargets.find(e => e.getAttribute("aria-selected") === "true") || null;
+        }
+        
+        /**
+         * Switches the tablist to the provided tab, updating the tabs and panels
+         * to reflect the change.
+         * @param selectedTab The tab to select. If `null` is provided or the element
+         * is not a valid tab, all tabs will be unselected.
+         */
+        public set selectedTab(selectedTab: HTMLElement | null) {
+            for (let tab of this.tabTargets) {
+                let panelId = tab.getAttribute('aria-controls');
+                let panel = panelId ? document.getElementById(panelId) : null;
+    
+                if (tab === selectedTab) {
+                    tab.classList.add('is-selected');
+                    tab.setAttribute('aria-selected', 'true');
+                    tab.removeAttribute('tabindex');
+                    panel?.classList.remove('d-none');
+                } else {
+                    tab.classList.remove('is-selected');
+                    tab.setAttribute('aria-selected', 'false');
+                    tab.setAttribute('tabindex', '-1');
+                    panel?.classList.add('d-none');
+                }
+            }
+        }
+    }
+}
+Stacks.application.register("s-navigation-tablist", Stacks.TabListController);
```

Comment:
nit: See above regarding `TabController`. I'd call this simply `s-tab` because it could be used for lots of different things.

Reply:
Same as above, the name here means it is intended to be used exclusively for a `s-navigation` element (because it applies things like `is-selected` rather than having something like `data-s-tablist-selection-class-name="is-selected"`) and because it is intended only for elements that are acting as a `tablist`.  Theoretically it could be used for a button group, but I would rather add `Stacks.application.register("s-btn-group-tablist", Stacks.TabListController);` as a second actor which allows for a potential subclass and differentiation than try making all of that property driven at this juncture.